src/core/public/chrome/ui/header/header.tsx

```diff
@@ -115,13 +116,24 @@ function extendRecentlyAccessedHistoryItem(
   };
 }
 
-function extendNavLink(navLink: ChromeNavLink) {
+function extendNavLink(navLink: ChromeNavLink, urlForApp: ApplicationStart['getUrlForApp']) {
+  if (navLink.legacy) {
+    return {
+      ...navLink,
+      href: navLink.url && !navLink.active ? navLink.url : navLink.baseUrl,
+    };
+  }
+
   return {
     ...navLink,
-    href: navLink.url && !navLink.active ? navLink.url : navLink.baseUrl,
+    href: urlForApp(navLink.id),
   };
 }
 
+function isModifiedEvent(event: MouseEvent) {
```

Comment:
no tests. I'm not sure that our testing framework supports `keyDown + click` tho